File: leetgo/go/LCR 141.fan-zhuan-lian-biao-lcof/solution.go
// Created by k at 2024/05/13 14:23
// leetgo: dev
// https://leetcode.cn/problems/fan-zhuan-lian-biao-lcof/

package main

import (
	"bufio"
	"fmt"
	"os"

	. "github.com/j178/leetgo/testutils/go"
)

// @lc code=begin

func trainningPlan(head *ListNode) (ans *ListNode) {
	var pre *ListNode
	for head != nil {
		next := head.Next
		head.Next = pre
		pre = head
		head = next
	}
	ans = pre
	return
}

// @lc code=end

func main() {
	stdin := bufio.NewReader(os.Stdin)
	head := Deserialize[*ListNode](ReadLine(stdin))
	ans := trainningPlan(head)

	fmt.Println("\noutput:", Serialize(ans))
}

File: leetgo/cpp/LCR 141.fan-zhuan-lian-biao-lcof/solution.cpp
// Created by k at 2024/05/13 14:35
// leetgo: dev
// https://leetcode.cn/problems/fan-zhuan-lian-biao-lcof/

#include <bits/stdc++.h>
#include "LC_IO.h"
using namespace std;

// @lc code=begin

class Solution {
  public:
    ListNode *trainningPlan(ListNode *head)
    {
        ListNode *pre = nullptr;
        for (; head != nullptr;) {
            ListNode *next = head->next;
            head->next     = pre;
            pre            = head;
            head           = next;
        }
        return pre;
    }
};

// @lc code=end

int main()
{
    ios_base::sync_with_stdio(false);
    stringstream out_stream;

    ListNode *head;
    LeetCodeIO::scan(cin, head);

    Solution *obj = new Solution();
    auto res      = obj->trainningPlan(head);
    LeetCodeIO::print(out_stream, res);
    cout << "\noutput: " << out_stream.rdbuf() << endl;

    delete obj;
    return 0;
}
